modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/footprint/SidecarFootprintProvider.java

```diff
@@ -219,7 +219,7 @@ private String getFullPath(String strValue) throws IOException {
         if (!file.isAbsolute()) {
             file = new File(reference, strValue);
         }
-        return file.getCanonicalPath();
+        return file.getPath();
```

Comment:
Can you explain this change?

Reply:
There is a test that struggles on Windows with constructing the full path. It was a mosaic test, I forgot the exact name. On line 220 in that test the reference was a file path to the target folder of the test e.g. c/windows/.../target/ and the strValue was something like http://localhost:9080/mosaic. So the file path became: c/windows/.../target/http://localhost:9080/mosaic which doesn't make any sense. The test data that was given, seemed ok at the right places. The file path it constructed wasn't really a valid path. When java tries to make it canonical (standardise) it, it breaks. I see no reason why someone would want to get the canonical version of the path. The only reason one would want to do that, is to compare file paths.

Reply:
@vjeetje this must be a bug in the code or a test. Nothing should be calling this method with the string encoding of a URL. @aaime is this a bug in `SidecarFootprintProvider`? Can the `location` be an HTTP URL?

Reply:
@bencaradocdavies before Niels changes to allow mosaic pulling from random URL definitely not, now I'm not so sure.

Reply:
@aaime @bencaradocdavies The test file that is containing the url which was parsed is: org/geotools/gce/imagemosaic/test-data/remote_test/remote_test.dbf . It contains the remote url  "http://localhost:8089/global_mosaic_0.png". Is this a stable feature that somehow works on linux?

Reply:
That was recently introduced in https://github.com/geotools/geotools/pull/1869 and already caused another regression fixed in https://github.com/geotools/geotools/pull/1910
It does not break the build on Linux, but don't believe it's actually working either, that code should probably have been amended to also work with URLs instead of files. I'm however not sure why it gets into the footprint management path...